src/a110x/a110x.cxx

```diff
@@ -44,6 +45,32 @@ A110X::A110X(int pin)
   m_isrInstalled = false;
 }
 
+A110X::A110X(std::string initStr) : mraaIo(initStr)
+{
+  mraa_io_descriptor* descs = mraaIo.getMraaDescriptors();
+  
+  if(!descs->gpios)
+  {
+    throw std::invalid_argument(std::string(__FUNCTION__) +
+                                  ": mraa_gpio_init() failed, invalid pin?");
+    return;
```

Comment:
return statement not required here. Exception will stop the code flow by itself.